src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildInfo.java

```diff
@@ -101,9 +101,20 @@ public void addBuilder(BuildBotData update) {
     builders.add(update);
   }
 
-  public boolean equals(BuildInfo other) {
-    return this.commitHash == other.commitHash && this.timestamp.equals(other.timestamp) &&
-           this.branch == other.branch && this.builders.equals(other.builders);
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
```

Comment:
consider adding blank lines between the if blocks for readability purposes

Reply:
I think it is already readable because these if blocks are very small